cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/transform/MockPipelineConfigurer.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.mock.transform;
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.module.DatasetModule;
+import co.cask.cdap.api.plugin.PluginProperties;
+import co.cask.cdap.api.plugin.PluginSelector;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.StageConfigurer;
+
+import javax.annotation.Nullable;
+
+/**
+ * Mock test configurer used to test configure pipeline's validation of input schema and setting of output shcema.
+ */
+public class MockPipelineConfigurer implements PipelineConfigurer {
```

Comment:
how many tests are using this one? Is it possible to refactor those tests rather than copying CoreValidator here and doing the hack for getPlugin() in this class? Would rather not introduce a dependency on commons-validator if possible.

If not, a better way to handle this would be to take a `Map<String, Object>` in the constructor as the plugins the configurer will return for getPlugin() instead of special casing one particular plugin. That way, it could be used for a variety of plugins if needed, and any dependencies remain local to whatever project is using it.